src/NzbDrone.Core/Tv/EpisodeRepository.cs

```diff
@@ -173,25 +175,44 @@ public void SetMonitoredBySeason(int seriesId, int seasonNumber, bool monitored)
             mapper.AddParameter("seasonNumber", seasonNumber);
             mapper.AddParameter("monitored", monitored);
 
-            const string sql = "UPDATE Episodes " +
-                               "SET Monitored = @monitored " +
-                               "WHERE SeriesId = @seriesId " +
-                               "AND SeasonNumber = @seasonNumber";
+            var sqlWhere = $"SeriesId = @seriesId AND SeasonNumber = @seasonNumber AND Monitored != @monitored";
+
+            var sqlGet = $"SELECT * FROM Episodes WHERE {sqlWhere}";
+
+            var episodes = mapper.Query<Episode>(sqlGet);
+            episodes.ForEach(v => v.Monitored = monitored);
+
+            mapper.AddParameter("seriesId", seriesId);
+            mapper.AddParameter("seasonNumber", seasonNumber);
+            mapper.AddParameter("monitored", monitored);
+
+            var sqlUpdate = $"UPDATE Episodes SET Monitored = @monitored WHERE {sqlWhere}";
 
-            mapper.ExecuteNonQuery(sql);
+            mapper.ExecuteNonQuery(sqlUpdate);
+
+            episodes.ForEach(ModelUpdated);
```

Comment:
Could avoid looping 2x and set monitored + send model updated here, but not that big of a concern.

Reply:
I can also just add ModelUpdate(id) coz it's gonna call GetResourceById anyway.